dids/src/main/kotlin/web5/sdk/dids/DidIonManager.kt

```diff
@@ -0,0 +1,353 @@
+    } else {
+      options.recoveryPublicJWK
+    }
+    val recoveryCommitment = publicKeyCommitment(recoveryPublicJWK)
+
+    val operation: OperationSuffixDataObject =
+      createOperationSuffixDataObject(createOperationDelta, recoveryCommitment)
+
+    return Pair(
+      SidetreeCreateOperation(
+        type = "create",
+        suffixData = operation,
+        delta = createOperationDelta,
+      ),
+      KeyAliases(
+        updateKeyAlias = updatePublicJWK.keyID,
+        verificationKeyAlias = verificationPublicKey.publicKeyJwk!!.keyID,
+        recoveryKeyAlias = recoveryPublicJWK.keyID
+      )
+    )
+  }
+
+  private fun validateVerificationMethodId(id: String) {
+    if (!isBase64UrlString(id)) {
+      throw IllegalArgumentException("verification method id \"$id\" is not base 64 url charset")
+    }
+
+    if (id.length > maxVerificationMethodIdLength) {
+      throw IllegalArgumentException(
+        "verification method id \"$id\" exceeds max allowed length of $maxVerificationMethodIdLength"
+      )
+    }
+  }
+
+  private fun isBase64UrlString(input: String?): Boolean {
+    val regex = "^[A-Za-z0-9_-]+$".toRegex()
+    return regex.matches(input!!)
+  }
+
+  private fun createOperationSuffixDataObject(
+    createOperationDeltaObject: Delta,
+    recoveryCommitment: String): OperationSuffixDataObject {
+    val jsonString = mapper.writeValueAsString(createOperationDeltaObject)
+    val canonicalized = JsonCanonicalizer(jsonString).encodedUTF8
+    val deltaHash = Multihash.sum(Multicodec.SHA2_256, canonicalized).get()?.bytes()
+    return OperationSuffixDataObject(
+      deltaHash = Base64URL.encode(deltaHash).toString(),
+      recoveryCommitment = recoveryCommitment
+    )
+  }
+
+  private fun publicKeyCommitment(publicKeyJWK: JWK): Commitment {
+    require(!publicKeyJWK.isPrivate)
+    // 1. Encode the public key into the form of a valid JWK.
+    val pkJson = publicKeyJWK.toJSONString()
+
+    // 2. Canonicalize the JWK encoded public key using the implementation’s JSON_CANONICALIZATION_SCHEME.
+    val canonicalized = JsonCanonicalizer(pkJson).encodedUTF8
+
+    // 3. Use the implementation’s HASH_PROTOCOL to Multihash the canonicalized public key to generate the REVEAL_VALUE,
+    val intermediate = Multihash.sum(Multicodec.SHA2_256, canonicalized).get()?.digest!!
+
+    // then Multihash the resulting Multihash value again using the implementation’s HASH_PROTOCOL to produce
+    // the public key commitment.
+    val hashOfHash = Multihash.sum(Multicodec.SHA2_256, intermediate).get()?.bytes()
+    return Base64URL.encode(hashOfHash).toString()
+  }
+
+  /**
+   * Default companion object for creating a [DidIonManager] with a default configuration.
+   */
+  public companion object Default : DidIonManager(DidIonConfiguration())
+}
+
+/**
+ * Represents an HTTP response where the status code is outside the range considered success.
+ */
+public class InvalidStatusException(s: String) : Exception(s)
```

Comment:
extends `RuntimeException`. Also given that it's a custom exception, would be nice to include http status code